pretyping/structures.ml

```diff
@@ -400,6 +400,34 @@ let is_open_canonical_projection env sigma c =
     with Failure _ -> false
   with Not_found -> false
 
+
+let decompose_reduced_canonical_projection sigma c =
```

Comment:
Did you check that the debruijn logic in this still works when letins are involved (`Record foo := { bla : t; baz := bli }`)? I don't know how letins and canonical projections interact.

Reply:
A new test case reveals that the answer is "no". Good catch!

Reply:
Should be fixed now. It was a simple confusion between `ci_cstr_nargs` and `ci_cstr_ndecls`. Reading the definition of these fields was enough to understand which one I should have used.